src/uu/shred/src/shred.rs

```diff
@@ -428,35 +430,37 @@ fn wipe_file(
 
     // Fill up our pass sequence
     let mut pass_sequence = Vec::new();
+    if metadata.len() != 0 {
+        // Only add passes if the file is non-empty
 
-    if n_passes <= 3 {
-        // Only random passes if n_passes <= 3
-        for _ in 0..n_passes {
-            pass_sequence.push(PassType::Random);
-        }
-    } else {
-        // First fill it with Patterns, shuffle it, then evenly distribute Random
-        let n_full_arrays = n_passes / PATTERNS.len(); // How many times can we go through all the patterns?
-        let remainder = n_passes % PATTERNS.len(); // How many do we get through on our last time through?
-
-        for _ in 0..n_full_arrays {
-            for p in PATTERNS {
-                pass_sequence.push(PassType::Pattern(p));
+        if n_passes <= 3 {
+            // Only random passes if n_passes <= 3
+            for _ in 0..n_passes {
+                pass_sequence.push(PassType::Random);
             }
-        }
-        for pattern in PATTERNS.into_iter().take(remainder) {
-            pass_sequence.push(PassType::Pattern(pattern));
-        }
-        let mut rng = rand::thread_rng();
-        pass_sequence.shuffle(&mut rng); // randomize the order of application
+        } else {
+            // First fill it with Patterns, shuffle it, then evenly distribute Random
+            let n_full_arrays = n_passes / PATTERNS.len(); // How many times can we go through all the patterns?
+            let remainder = n_passes % PATTERNS.len(); // How many do we get through on our last time through?
+
+            for _ in 0..n_full_arrays {
+                for p in PATTERNS {
+                    pass_sequence.push(PassType::Pattern(p));
+                }
+            }
+            for pattern in PATTERNS.into_iter().take(remainder) {
+                pass_sequence.push(PassType::Pattern(pattern));
+            }
+            let mut rng = rand::thread_rng();
+            pass_sequence.shuffle(&mut rng); // randomize the order of application
 
-        let n_random = 3 + n_passes / 10; // Minimum 3 random passes; ratio of 10 after
-                                          // Evenly space random passes; ensures one at the beginning and end
-        for i in 0..n_random {
-            pass_sequence[i * (n_passes - 1) / (n_random - 1)] = PassType::Random;
+            let n_random = 3 + n_passes / 10; // Minimum 3 random passes; ratio of 10 after
+                                              // Evenly space random passes; ensures one at the beginning and end
+            for i in 0..n_random {
+                pass_sequence[i * (n_passes - 1) / (n_random - 1)] = PassType::Random;
+            }
         }
     }
-
     // --zero specifies whether we want one final pass of 0x00 on our file
     if zero {
         pass_sequence.push(PassType::Pattern(PATTERNS[0]));
```

Comment:
I think this condition should also be inside the `if metadata.len() != 0` block.

Reply:
sorry, i missed your comment. you are correct ; well spotted!